erpnext/stock/doctype/item/item.py

```diff
@@ -470,6 +472,13 @@ def set_shopping_cart_data(self, context):
 		from erpnext.shopping_cart.product_info import get_product_info_for_website
 		context.shopping_cart = get_product_info_for_website(self.name, skip_quotation_creation=True)
 
+	def set_page_blocks(self, context):
```

Comment:
I dont like this duplication at all. Atleast, we should extract the Page Builder parts of the code and put it in a class and share that class among "Page Builder"able doctypes.